### PR TITLE
Update build for Apache releases

### PR DESCRIPTION
**Pull request by rdblue**
* Add Apache publication
* Add source, javadoc, and test artifacts with signatures to publication
* Add Apache snapshot and release repositories
* Add metadata to publication POM files
* Add missing licenses and update RAT excludes
* Add DISCLAIMER
* Add disclaimer page to ASF site
* Add release scripts

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/PartitionSpec.java [hunk 1]
```diff
@@ -482,6 +482,8 @@ public PartitionSpec build() {
   static void checkCompatibility(PartitionSpec spec, Schema schema) {
     for (PartitionField field : spec.fields) {
       Type sourceType = schema.findType(field.sourceId());
+      ValidationException.check(sourceType != null,
+          "Cannot find source column for partition field: %s", field);
```
Comment (xabriel) [inline api/src/main/java/org/apache/iceberg/PartitionSpec.java]: Looks unrelated to Apache release?

Comment (rdblue) [reply api/src/main/java/org/apache/iceberg/PartitionSpec.java]: Yeah, looks like this was included by accident. I'll open a separate PR for it.
